cmd/cluster-controller/main.go

```diff
@@ -2,112 +2,78 @@ package main
 
 import (
 	"context"
-	"flag"
+	"errors"
 	"os"
 	"os/signal"
 	"time"
 
+	"github.com/spf13/pflag"
+
 	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	crdexternalversions "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog/v2"
-	"k8s.io/kubernetes/pkg/genericcontrolplane/clientutils"
 
 	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
 	kcpexternalversions "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
-	"github.com/kcp-dev/kcp/pkg/reconciler/apiresource"
 	"github.com/kcp-dev/kcp/pkg/reconciler/cluster"
 )
 
-const numThreads = 2
 const resyncPeriod = 10 * time.Hour
 
-var (
-	kubeconfigPath  = flag.String("kubeconfig", "", "Path to kubeconfig")
-	syncerImage     = flag.String("syncer_image", "", "Syncer image to install on clusters")
-	pullMode        = flag.Bool("pull_mode", true, "Deploy the syncer in registered physical clusters in POD, and have it sync resources from KCP")
-	pushMode        = flag.Bool("push_mode", false, "If true, run syncer for each cluster from inside cluster controller")
-	autoPublishAPIs = flag.Bool("auto_publish_apis", false, "If true, the APIs imported from physical clusters will be published automatically as CRDs")
-)
+func bindOptions(fs *pflag.FlagSet) *options {
```

Comment:
Thoughts on making this a method on options and having it create & return a flagset? (This is what I've seen elsewhere). I don't know that I have strong feelings one way or another - mostly just comparing w/other art I'm familiar with...

Reply:
Passing in the flagset allows for composition, so it's useful in the helper library for e.g. the `cluster` config. I just used the same approach here so it would be parallel & simple, but yeah I guess it doesn't matter one way or the other in `main`

Reply:
I think I might prefer it this way so that `bind` has one meaning across the codebase?